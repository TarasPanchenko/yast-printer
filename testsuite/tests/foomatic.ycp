/**
 * File:
 *  foomatic.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Foomatic data manipulation tests
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);
    import "Pkg";
    import "Printer";
    include "printer/misc.ycp";
    import "Mode";
    TESTSUITE_CLEAN ();

    Mode::test = true;

    map READ = $[
	"ppd" : $[
	    "db" : $[
		"vendorname" : "EPSON",
		"modelname" : "STYLUSPHOTO810",
		"changed" : false,
	    ]
	],
	"target" : $[
	    "ycp" : $["__default__": [ ".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
	]
    ];

    DUMP ("========================================");

    TEST (``(getFoomaticDefaultQueue ([$["name" : "queue1"], $["name" : "queue2"]])), [], 0);
    TEST (``(getFoomaticDefaultQueue ([$["name" : "queue1"], $["name" : "normal"]])), [], 0);
    TEST (``(getFoomaticDefaultQueue ([$["name" : "queue1"], $["name" : "normal3"]])), [], 0);
    TEST (``(getFoomaticDefaultQueue ([$["name" : "queue1"], $["name" : "normal_gray"]])), [], 0);
    TEST (``(getFoomaticDefaultQueue ([$["name" : "normal"], $["name" : "normal2"]])), [], 0);
    TEST (``(getFoomaticDefaultQueue ([$["name" : "normal1"], $["name" : "normal"]])), [], 0);

    DUMP ("========================================");

    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["P1 (recommended)", "P2"])), [READ, $[], $[]], 0);
    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["P1", "P2"])), [READ, $[], $[]], 0);
    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["P1", "P2 (recommended)"])), [READ, $[], $[]], 0);

    DUMP ("========================================");

    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY", "ZZpcl3ZZ", "QQcdj3QQ"])), [READ, $[], $[]], 0);
    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY", "ZZpcl3ZZ (recommended)", "QQcdj3QQ"])), [READ, $[], $[]], 0);
    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY (recommended)", "ZZpcl3ZZ", "QQcdj3QQ  (recommended)"])), [READ, $[], $[]], 0);
    TEST (``(sortPpds ("EPSON", "Stylus Photo 810", ["XXgimpXX (recommended)", "YYhpijsYY (recommended)", "ZZpcl3ZZ (recommended)", "QQcdj3QQ (recommended)"])), [READ, $[], $[]], 0);

    DUMP ("========================================");

    Printer::spooler = "cups";

    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["P1 (recommended)", "P2"])), [READ, $[], $[]], 0);
    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["P1", "P2"])), [READ, $[], $[]], 0);
    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["P1", "P2 (recommended)"])), [READ, $[], $[]], 0);

    DUMP ("========================================");

    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY", "ZZpcl3ZZ", "QQcdj3QQ"])), [READ, $[], $[]], 0);
    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY", "ZZpcl3ZZ (recommended)", "QQcdj3QQ"])), [READ, $[], $[]], 0);
    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["XXgimpXX", "YYhpijsYY (recommended)", "ZZpcl3ZZ", "QQcdj3QQ  (recommended)"])), [READ, $[], $[]], 0);
    TEST (``(getAutoPpdFile ("EPSON", "Stylus Photo 810", ["XXgimpXX (recommended)", "YYhpijsYY (recommended)", "ZZpcl3ZZ (recommended)", "QQcdj3QQ (recommended)"])), [READ, $[], $[]], 0);

    DUMP ("========================================");



}
