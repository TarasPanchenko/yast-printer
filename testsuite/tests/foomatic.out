Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Dump	========================================
Return	queue1
Return	normal
Return	normal3
Return	queue1
Return	normal2
Return	normal
Dump	========================================
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Execute	.target.bash "/bin/grep '#this database was generated during package building' /var/lib/YaST2/ppd_db.ycp" 0
Read	.ppd.db.changed false
Read	.target.ycp "/var/lib/YaST2/ppd_db.ycp" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["P1 (recommended)", "P2"]
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["P1", "P2"]
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["P2 (recommended)", "P1"]
Dump	========================================
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["YYhpijsYY", "XXgimpXX", "QQcdj3QQ", "ZZpcl3ZZ"]
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["ZZpcl3ZZ (recommended)", "YYhpijsYY", "XXgimpXX", "QQcdj3QQ"]
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["YYhpijsYY (recommended)", "QQcdj3QQ  (recommended)", "XXgimpXX", "ZZpcl3ZZ"]
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	["YYhpijsYY (recommended)", "XXgimpXX (recommended)", "QQcdj3QQ (recommended)", "ZZpcl3ZZ (recommended)"]
Dump	========================================
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	P1 (recommended)
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	P1
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	P2 (recommended)
Dump	========================================
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	YYhpijsYY
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	ZZpcl3ZZ (recommended)
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	YYhpijsYY (recommended)
Read	.ppd.db.vendorname "EPSON" "EPSON"
Read	.ppd.db.modelname ["EPSON", "Stylus Photo 810"] "STYLUSPHOTO810"
Execute	.target.bash "/usr/bin/touch /usr/share/YaST2/data/printer/ppd_preference_list" 0
Read	.target.ycp "/usr/share/YaST2/data/printer/ppd_preference_list" $["__default__":[".*hpijs.*", ".*gimp.*", ".*escp.*", ".*upp.*", ".*cdj[0-9].*", ".*pcl3.*"]]
Return	YYhpijsYY (recommended)
Dump	========================================
